Simplified/Catalog/NYPLCatalogNavigationController.m

```diff
@@ -194,47 +194,28 @@ - (void)viewDidAppear:(BOOL)animated
   // TODO: SIMPLY-3048 refactor better in a extension
 #ifdef SIMPLYE
   NYPLSettings *settings = [NYPLSettings sharedSettings];
-  
-  if (!settings.userHasSeenWelcomeScreen) {
-    Account *currentAccount = [[AccountsManager sharedInstance] currentAccount];
-
-    __block NSURL *mainFeedUrl = [NSURL URLWithString:currentAccount.catalogUrl];
-    void (^completion)(void) = ^() {
-      [[NYPLSettings sharedSettings] setAccountMainFeedURL:mainFeedUrl];
-      [UIApplication sharedApplication].delegate.window.tintColor = [NYPLConfiguration mainColor];
-      
-      NYPLWelcomeScreenViewController *welcomeScreenVC = [[NYPLWelcomeScreenViewController alloc] initWithCompletion:^(Account *const account) {
-        if (![NSThread isMainThread]) {
```

Comment:
this check is unneeded because this inner completion is always called on the main thread, according to `NYPLWelcomeScreenViewController`'s code.